esphome/components/opentherm/opentherm.h

```diff
@@ -0,0 +1,290 @@
+#pragma once
+
+#include "esphome/core/helpers.h"
+#include "esphome/core/defines.h"
+#include "esphome/core/component.h"
+#include "esphome/core/gpio.h"
+#include "esphome/core/hal.h"
+#ifdef USE_SENSOR
+#include "esphome/components/sensor/sensor.h"
+#endif
+#ifdef USE_BINARY_SENSOR
+#include "esphome/components/binary_sensor/binary_sensor.h"
+#endif
+#ifdef USE_SWITCH
+#include "switch/custom_switch.h"
+#endif
+#ifdef USE_NUMBER
+#include "number/custom_number.h"
+#endif
+#include "consts.h"
+#include "enums.h"
+#include <queue>
+
+namespace esphome {
+namespace opentherm {
+
+class OpenThermComponent : public PollingComponent {
+ public:
+#ifdef USE_SENSOR
+  sensor::Sensor *ch_min_temperature_sensor_{nullptr};
```

Comment:
All of the sensors etc should be `protected` not `public`
A macro exists to make this a little easier and defines the pointer and the setter for you.

```suggestion
  SUB_SENSOR(ch_min_temperature)
```